seagoat/sources/chroma.py

```diff
@@ -87,28 +142,138 @@ def fetch(query_text: str, limit: int):
 
         return format_results(query_text, repository, chromadb_results)
 
-    def cache_chunk(chunk):
+    def _add_chunks_to_collection(ids: List[str], documents: List[str], metadatas: List[Dict]) -> List[int]:
+        """
+        Add chunks to collection with better error handling.
+        Returns list of indices that failed to add.
+        """
+        failed_indices = []
+        
         try:
+            # Try to add all chunks at once
             chroma_collection.add(
```

Comment:
as I can see, this is only really being called to handle chunks that could not be added due to some errors. I might be wrong here, but my intuition says that there is no point in trying to add them all ot once, as most likely they cannot be added. I don't remember seeing any "flaky" behavior from ChromaDB